File: app/src/main/java/com/nuhkoca/offlineapp/ui/news/NewsViewModel.java
package com.nuhkoca.offlineapp.ui.news;

import com.nuhkoca.offlineapp.data.local.entity.News;
import com.nuhkoca.offlineapp.repository.Resource;

import java.util.List;

import javax.inject.Inject;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

public class NewsViewModel extends ViewModel {
    private final LiveData<Resource<List<News>>> news;

    @Inject
    NewsViewModel(@NonNull NewsUseCase newsUseCase) {
        news = newsUseCase.execute(null);
    }

    public LiveData<Resource<List<News>>> getNews() {
        return news;
    }
}

File: app/src/main/java/com/nuhkoca/offlineapp/ui/news/NewsViewModel.kt
package com.nuhkoca.offlineapp.ui.news

import com.nuhkoca.offlineapp.data.local.entity.News
import com.nuhkoca.offlineapp.repository.Resource

import javax.inject.Inject
import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.nuhkoca.offlineapp.repository.UseCaseLiveData

class NewsViewModel @Inject
internal constructor(newsUseCase: NewsUseCase) : ViewModel() {
    val news: LiveData<Resource<List<News>>> = newsUseCase.execute(UseCaseLiveData.None())
}
